Synergy/Controllers/TeamsController.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Synergy.Authorization;
+using Synergy.Models;
+using Synergy.Services;
+using Permissions = Synergy.Authorization.Permissions;
+
+namespace Synergy.Controllers;
+
+[ApiController]
+[Route("/api/[controller]")]
+[Authorize]
+public class TeamsController : ControllerBase
+{
+    private readonly TeamsService _teamsService;
+    private readonly UsersService _usersService;
+
+    public TeamsController(TeamsService teamsService, UsersService usersService)
+    {
+        _teamsService = teamsService;
+        _usersService = usersService;
+    }
+
+    [AuthorizePermission(Permissions.CanRegisterTeam)]
+    [HttpPost("register-team")]
+    public async Task<IActionResult> RegisterTeam([FromBody] RegisterTeamBody body)
+    {
+        var team = body.Team;
```

Comment:
Going to need to show you FluentValidation because you have no handling for if these values are null.